### PR TITLE
Move to standard folder and update game loader

### PR DESCRIPTION
**Pull request by DanVanAtta**
Included:
- move source files to standard folder locations: https://github.com/triplea-game/triplea/issues/1435
- move tutorial.zip out of a 'maps' folder, push it up by one. Avoiding a 'maps' folder bundled with the game will hopefully reduce confusion on where maps can/should be. 
- remove DTD parsing for maps in test. This is only done in test, mainly just adding headaches
- a number of simplifications

New folder layout:
new_folder_layout

### CODE REVIEW

**Comment by ron-murhammer** [comment]
@DanVanAtta Isn't this gonna conflict pretty hard with all other PRs and work in progress?

**Comment by DanVanAtta** [comment]
Git should be able to resolve most of them automatically.

**Comment by DanVanAtta** [comment]
Seems things are working well now. Ready for a review.

**Comment by simon33-2** [comment]
Why?

**Comment by DanVanAtta** [comment]
@simon33-2  why what?

@ron-murhammer please take a look

**Comment by ron-murhammer** [comment]
@DanVanAtta This looks to have a number of conflicts. My guy tells me it might be better to either redo the changes on the current master or wait til before a large release to do this since it'll conflict with lots of other PRs.

**Comment by DanVanAtta** [comment]
@ron-murhammer  I do not want to whack-a-mole resolve conflicts to then have more come along. Please review what has been changed and I'll take a crack at the conflicts and any review comments in one go.

File: src/main/java/games/strategy/engine/framework/GameRunner.java [hunk 1]
```diff
@@ -346,12 +346,6 @@ private static String getValue(final String arg) {
 
   public static void setupLogging(GameMode gameMode) {
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/framework/GameRunner.java]: Probably would have been better to put logging changes in a different PR.

File: src/main/java/games/strategy/engine/framework/GameRunner.java [hunk 2]
```diff
@@ -346,12 +346,6 @@ private static String getValue(final String arg) {
...
     if (gameMode == GameMode.SWING_CLIENT) {
-      // setup logging to read our logging.properties
-      try {
-        LogManager.getLogManager().readConfiguration(ClassLoader.getSystemResourceAsStream("logging.properties"));
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/framework/GameRunner.java]: Should the properties file be removed then?

File: src/main/java/games/strategy/engine/framework/GameRunner.java [hunk 3]
```diff
@@ -364,17 +358,6 @@ protected void dispatchEvent(AWTEvent newEvent) {
           }
         }
       });
-    } else {
-      // setup logging to read our logging.properties
-      try {
-        LogManager.getLogManager()
-            .readConfiguration(ClassLoader.getSystemResourceAsStream("headless-game-server-logging.properties"));
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/framework/GameRunner.java]: Should the properties file be removed then?

File: src/main/java/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java [hunk 4]
```diff
@@ -656,36 +656,12 @@ public Chat getChat() {
...
+    GameRunner.handleCommandLineArgs(args, getProperties(), GameRunner.GameMode.HEADLESS_BOT);
     ClipPlayer.setBeSilentInPreferencesWithoutAffectingCurrent(true);
-    HeadlessGameServer server = null;
     try {
-      server = new HeadlessGameServer();
+      new HeadlessGameServer();
     } catch (final Exception e) {
-      ClientLogger.logQuietly(e);
-    }
-    if (Boolean.parseBoolean(System.getProperty(GameRunner.TRIPLEA_GAME_HOST_CONSOLE_PROPERTY, "false"))) {
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java]: What was this used for and are we sure we can remove it?

File: src/main/java/games/strategy/engine/data/GameParser.java [hunk 5]
```diff
@@ -247,15 +247,15 @@ private void validateAttachments(final Attachable attachable) throws GameParseEx
...
     final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
-    factory.setValidating(true);
-    // get the dtd location
-    final String dtdFile = "/games/strategy/engine/xml/" + DTD_FILE_NAME;
-    final URL url = GameParser.class.getResource(dtdFile);
-    if (url == null) {
-      throw new RuntimeException("Map: " + mapName + ", " + String.format("Could not find in classpath %s", dtdFile));
-    }
-    final String dtdSystem = url.toExternalForm();
-    final String system = dtdSystem.substring(0, dtdSystem.length() - 8);
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/data/GameParser.java]: Why are we removing the DTD?

File: src/test/java/games/strategy/triplea/xml/LoadGameUtil.java [hunk 6]
```diff
@@ -30,14 +43,22 @@
     public String toString() {
       return value;
     }
+
+    private InputStream getInputStream() throws IOException {
+      File f = new File(TEST_MAP_XML_PATH + value);
+      System.out.println("f .... => " + f.getAbsolutePath());
```
Comment (ron-murhammer) [inline src/test/java/games/strategy/triplea/xml/LoadGameUtil.java]: Should we remove these system.out.println's?

**Comment by ron-murhammer** [comment]
Should the 'data' classes be moved to just 'src/main/resources' rather than 'src/main/resources/data'?

File: src/main/java/games/strategy/engine/data/GameParser.java (hunk 5 above)
Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/data/GameParser.java]: It's only used in test. That is not helpful since the real ones can be arbitrary.

**Comment by ron-murhammer** [comment]
@DanVanAtta Added a few comments/questions. Overall though, looks good. Just concerned about accidentally overriding previous changes if you rebase this. We need to be very careful if you go that route.

**Comment by DanVanAtta** [comment]
> " 'src/main/resources' rather than 'src/main/resources/data'?"

I would prefer `resources/data` for the moment. It makes the installer updates much more straight forward, reduces the scope/risk of this update.

**Comment by ssoloff** [comment]
> Just concerned about accidentally overriding previous changes if you rebase this. We need to be very careful if you go that route.

@DanVanAtta, @ron-murhammer: I'd be happy to review some of the rebased changes, as a good number of the conflicts are in files I recently changed.  Hopefully, it will be more obvious to me if a merge conflict is not resolved correctly in those files.

File: src/main/java/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java (hunk 4 above)
Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java]: TRIPLEA_GAME_HOST_CONSOLE_PROPERTY triggers whether we show the bot UI. We have agreed we will stop launching bots in screen sessions and will no longer be using their UI. The majority of bots are not running with this configuration.

File: src/main/java/games/strategy/engine/framework/GameRunner.java (hunk 1 above)
Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/framework/GameRunner.java]: Somewhere down the rabbit hole I think there was a dependency that required these updates. Otherwise perhaps so, moot point I think.

File: src/main/java/games/strategy/engine/framework/GameRunner.java (hunk 2 above)
Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/framework/GameRunner.java]: removed, did not spot any other usages

File: src/main/java/games/strategy/engine/framework/GameRunner.java (hunk 3 above)
Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/framework/GameRunner.java]: Same, also removed.

**Comment by DanVanAtta** [comment]
@ssoloff  thanks for the other, though the merge conflict was very easy, just had to confirm that a single file was to remain deleted.

@ron-murhammer  please have another look and let me know if there is anything else further to address.
